### DISCUSSION TITLE
How can i connect Google Shets to Supabase

### DISCUSSION THREAD

**Discussion by debanath** [supabase/supabase#23624] [Questions]
Lets say, i want to connect my google sheets to the Supabase DB using the google appscript. Also i dont want the data to be updated regularly i only want the data to be updated once a month becuase of which i dont want any automation there i am willing to run the function manually anyways because i'll be updating the values myself. I tried using JDBC suggested by "ChatGPT" fbut for some reason i got   
```
Execution log
11:48:24 AM	Notice	Execution started
11:48:24 AM	Info	Error connecting to database: Exception: Connection URL uses an unsupported JDBC protocol.
11:48:25 AM	Notice	Execution completed
```
for the code

```js
function transferDataToDatabase() {
  var sheet = SpreadsheetApp.getActiveSpreadsheet().getActiveSheet();
  var data = sheet.getDataRange().getValues();

  // Set up the connection string
  var connectionString = '<myjdbc_connectionString_from_Supabase>';

  var connection = null;
  try {
    connection = Jdbc.getConnection(connectionString);

    // Iterate over the data and insert it into the database
    for (var i = 1; i < data.length; i++) { // Start from 1 to skip headers
      var row = data[i];

      // Assuming the table name is in the first column and other columns represent data
      var tableName = row[0];
      var rowData = row.slice(1); // Exclude the table name from data
      
      insertDataIntoDatabase(tableName, rowData, connection);
    }
  } catch (e) {
    Logger.log('Error connecting to database: ' + e);
  } finally {
    if (connection != null) {
      connection.close();
    }
  }
}

function insertDataIntoDatabase(tableName, rowData, connection) {
  var statement = connection.prepareStatement('INSERT INTO ' + tableName + ' VALUES (?)');
  statement.setString(1, rowData.join(','));
  statement.executeUpdate();
  statement.close();
}
```
If someone is aware of what i am doing wrong and willing to help me with some suggestion Please feel free to do so.
Thank you in Advance

**Comment by MakeMeDizzy** [comment]
@debanath
It depends which direction you need — let's split it:

Just pull Supabase data into a Sheet (read-only) — easiest:
In Supabase: Settings → API, copy your project URL + API key.
In your Sheet: Extensions → Apps Script.
Use UrlFetchApp.fetch() to call https://<project>.supabase.co/rest/v1/<your_table> and write the returned JSON rows into the sheet.
Add a time-driven trigger so it refreshes on a schedule.

Two-way (edit in the sheet → write back to the DB) — this is where it gets tricky for a first project: Sheets can't notify your database automatically, so you have to poll, and you need to decide what happens when the same row changed in both places.
If you'd rather not write and babysit a script, I'm building a tool that handles the two-way part for you — pick a table, pick a sheet, choose a conflict rule, done. Early access: https://glistening-palmier-d44c63.netlify.app/?utm_source=gh_23624
Tell me which direction you actually need and I'm happy to point you the right way.
